pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -498,16 +499,17 @@ func (s *DiscoveryServer) DeltaAggregatedResources(stream ads.AggregatedDiscover
 // for large configs. The method will hold a lock on con.pushMutex.
 func (s *DiscoveryServer) pushConnection(con *XdsConnection, pushEv *XdsEvent) error {
 	// TODO: update the service deps based on NetworkScope
+	if !pushEv.full {
+		edsUpdatedServices := pushEv.configsUpdated[model.ServiceEntryKind]
```

Comment:
I don't understand how we distinguish between a Service changing and Endpoints changing if we are putting endpoints under the Service key? We used to do this by having a separate field for endpoints

Reply:
@howardjohn We did not do that when Service changing. I think i know what you mean. In a further step, we may need to distinguish whether it is service or service endpoint change.  Luckily we have added a new api `WorkloadEntry`. IMO, we should mark it `WorkloadEntry` when endpoint updated instead of `ServiceEntry` used now.

Reply:
Yeah, I agree with @hzxuzhonghu. The map has left extension point. If we do want to distinguish the two changes. We just need to add another "Entry". This makes the code more coherent.

Reply:
I haven't looked at this PR in detail and I am not fully following this conversation - but the distinction we made for service entries when actual service changed vs change endpoints for triggering full push vs just eds is not affected by this change? Is that correct?

Reply:
Hi~The push logic is exactly what it used to be. This PR only introduced code refactoring, nothing added. For more details:
* actual service changed -> full flag will be set "true" and ServiceEntryKind will present in ConfigsUpdated with empty "eds" resources names, so a full push will be trigged
* change endpoints for triggering full -> full flag will be set "true" and ServiceEntryKind will present in ConfigsUpdated with empty "eds" resources names, so a full push will be trigged
*  just eds-> full flag will be set "false
 and ServiceEntryKind will present in ConfigsUpdated with changed "eds" names, so an eds push will be trigged

If we encountered any full push while merging, the "eds" names will be eliminated and will be set to full push.

Reply:
Ok. Thank you for the confirmation

Reply:
As @howardjohn worries. If we really need to distinguished 1&2 in above situations. We just mark the "change endpoints" with "WorkloadEntry", not "ServiceEntryKind". But currently it is "ServiceEntryKind".